clusterloader2/pkg/measurement/common/generic_query_measurement.go

```diff
@@ -68,39 +104,58 @@ func (g *genericQueryGatherer) IsEnabled(config *measurement.Config) bool {
 }
 
 func (g *genericQueryGatherer) Gather(executor QueryExecutor, startTime, endTime time.Time, config *measurement.Config) ([]measurement.Summary, error) {
```

Comment:
Can we add some unit tests for this method? They can simply use fakeQueryExecutor to returns some artificial samples. The goal is to validate:
* that threshold is properly validated (no threshold, threshold higher than value, threshold lower than value)
* that the conversion to the summary happens as we want
* what happens if there are len(samples) > 1
* what happens when len(samples) == 0

Reply:
Done